doc/source/whatsnew/v0.23.2.txt

```diff
@@ -83,4 +83,5 @@ Bug Fixes
 **Other**
 
 - Bug in :meth:`Series.nlargest` for signed and unsigned integer dtypes when the minimum value is present (:issue:`21426`)
--
+- Bug in :class:`_Concatenator` should maintain dict ordering when :meth:`concat` is called (:issue:`2151`)
+-
```

Comment:
this needs to go into 0.24.0 as even though its a bug fix, its a jaring one, should be in reshaping.